### PR TITLE
update readme

### PR DESCRIPTION
**Pull request by brotzeit**
I've tried to reduce the contents to the user relevant stuff and put the porting information in the cookbook. Except for the blog post of @db48x.

@Wilfred I'm not sure which tasks should be listed, but I think this was pretty old.

We can put tasks on the list that seem relevant at this point and maybe focus on the remaining lisp functions. I would for example put `expand-file-name` on the list, since it's a fairly big function and there could be a rust crate that handles most of it for us.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -244,69 +238,43 @@ You can then open these docs with:
...
 
-Let's look at porting `numberp` to Rust.
+The first thing to look at is the C implementation for the atan function. It takes an optional second argument, which makes it interesting. The complicated mathematical bits, on the other hand, are handled by the standard library. This allows us to focus on the porting process without getting distracted by the math.
 
-First, make sure you have configured and built Remacs on your
-system. You'll probably want to generate TAGS too, so you can jump to
-definitions of C functions.
-
-This is the definition of `numberp`:
+The Lisp values we are given as arguments are tagged pointers; in this case they are pointers to doubles. The code has to check the tag and follow the pointer to retrieve the real values. Note that this code invokes a C macro (called DEFUN) that reduces some of the boilerplate. The macro declares a static varable called Satan that holds the metadata the Lisp compiler will need in order to successfully call this function, such as the docstring and the pointer to the Fatan function, which is what the C implementation is named:
```
Comment (shaleh) [inline README.md]: 'Satan' should have backticks.

File: README.md [hunk 2]
```diff
@@ -244,69 +238,43 @@ You can then open these docs with:
...
-        // … etc
+    {
+      double d2 = extract_float (x);
+      d = atan2 (d, d2);
     }
+  return make_float (d);
 }
 ```
 
-Similarly, `lisp_fn` can automatically translate the return type:
```
Comment (shaleh) [inline README.md]: Here too. Look for places to add backticks.

File: README.md [hunk 3]
```diff
@@ -244,69 +238,43 @@ You can then open these docs with:
...
+If only one argument Y is given, return the inverse tangent of Y.
+If two arguments Y and X are given, return the inverse tangent of Y
+divided by X, i.e. the angle in radians between the vector (X, Y)
+and the x-axis.  */)
+  (Lisp_Object y, Lisp_Object x)
 {
-  if (NUMBERP (object))
-    return Qt;
+  double d = extract_float (y);
+f (NILP (x))
```
Comment (Wilfred) [inline README.md]: Looks like this should be `if`?

**Comment by brotzeit** [comment]
Done.

File: README.md [hunk 4]
```diff
@@ -244,69 +238,43 @@ You can then open these docs with:
 $ cargo doc --open
 ```
 
-## Porting Elisp Primitive Functions: Walkthrough
+## Porting Elisp Primitive Functions
 
-Let's look at porting `numberp` to Rust.
```
Comment (shaleh) [inline README.md]: s/atan/`atan`/

**Comment by brotzeit** [comment]
I'm sure I will find some more stuff the next days. I will make a TODO list and open a new PR.

**Comment by shaleh** [comment]
Sounds good.

**Pull request merged**